pkg/pipelineascode/pipelineascode.go

```diff
@@ -69,99 +81,145 @@ func Run(ctx context.Context, cs *params.Run, providerintf provider.Interface, k
 	}
 
 	if !allowed {
-		msg := fmt.Sprintf("User %s is not allowed to run CI on this repo.", event.Sender)
-		cs.Clients.Log.Info(msg)
-		if event.AccountID != "" {
-			msg = fmt.Sprintf("User: %s AccountID: %s is not allowed to run CI on this repo.", event.Sender,
-				event.AccountID)
+		msg := fmt.Sprintf("User %s is not allowed to run CI on this repo.", p.event.Sender)
+		p.run.Clients.Log.Info(msg)
+		if p.event.AccountID != "" {
+			msg = fmt.Sprintf("User: %s AccountID: %s is not allowed to run CI on this repo.", p.event.Sender, p.event.AccountID)
 		}
 		status := provider.StatusOpts{
 			Status:     "completed",
 			Conclusion: "skipped",
 			Text:       msg,
 			DetailsURL: "https://tenor.com/search/police-cat-gifs",
 		}
-		if err := providerintf.CreateStatus(ctx, event, cs.Info.Pac, status); err != nil {
-			return fmt.Errorf("failed to run create status, user is not allowed to run: %w", err)
+		if err := p.vcx.CreateStatus(ctx, p.event, p.run.Info.Pac, status); err != nil {
+			return nil, nil, fmt.Errorf("failed to run create status, user is not allowed to run: %w", err)
 		}
-		return nil
+		return nil, nil, nil
 	}
 
-	pipelineRuns, err := getAllPipelineRuns(ctx, cs, providerintf, event)
+	pipelineRuns, err := getAllPipelineRuns(ctx, p.run, p.vcx, p.event)
 	if err != nil {
-		return err
+		return nil, nil, err
 	}
+
 	if pipelineRuns == nil {
-		msg := fmt.Sprintf("could not find templates in %s/ directory for this repository in %s", tektonDir, event.HeadBranch)
-		cs.Clients.Log.Info(msg)
-		return nil
+		msg := fmt.Sprintf("cannot locate templates in %s/ directory for this repository in %s", tektonDir, p.event.HeadBranch)
+		p.run.Clients.Log.Info(msg)
+		return nil, nil, nil
 	}
 
-	// Match the pipelinerun with annotation
-	pipelineRun, annotationRepo, config, err := matcher.MatchPipelinerunByAnnotation(ctx, pipelineRuns, cs, event)
+	// Match the PipelineRun with annotation
+	matchedPRs, err := matcher.MatchPipelinerunByAnnotation(ctx, pipelineRuns, p.run, p.event)
 	if err != nil {
 		// Don't fail when you don't have a match between pipeline and annotations
-		cs.Clients.Log.Warn(err.Error())
-		return nil
+		// TODO: better reporting
+		p.run.Clients.Log.Warn(err.Error())
+		return nil, nil, nil
+	}
+
+	return matchedPRs, repo, nil
+}
+
+func (p *PacRun) Run(ctx context.Context) error {
+	matchedPRs, repo, err := p.matchRepoPR(ctx)
+	if err != nil {
+		createStatusErr := p.vcx.CreateStatus(ctx, p.event, p.run.Info.Pac, provider.StatusOpts{
+			Status:     "completed",
+			Conclusion: "failure",
+			Text:       fmt.Sprintf("There was an issue validating the commit: %q", err),
+			DetailsURL: p.run.Clients.ConsoleUI.URL(),
+		})
+		if createStatusErr != nil {
+			p.run.Clients.Log.Errorf("Cannot create status: %s %s", err, createStatusErr)
+		}
 	}
 
-	if annotationRepo.Spec.URL != "" {
-		repo = annotationRepo
+	// TODO: We need to figure out secretCreation, it's buggy normally without multiplexing ie:bug #543
```

Comment:
what if we create one secret for each pipeline?
anyway the secret is short lived till the life of pr

Reply:
till we find a way to enable this again what would be side effect of this change?
multiple pipeline from private repo would fail right?

Reply:
let us work on this in a following PR,  i may have a nice solution,